nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncodeContent.java

```diff
@@ -29,55 +37,88 @@
 public class EncodeContent extends AbstractProcessor {
 
-    public static final String ENCODE_MODE = "Encode";
-    public static final String DECODE_MODE = "Decode";
-
-    public static final String BASE64_ENCODING = "base64";
-    public static final String BASE32_ENCODING = "base32";
-    public static final String HEX_ENCODING = "hex";
-
     public static final PropertyDescriptor MODE = new PropertyDescriptor.Builder()
             .name("Mode")
             .description("Specifies whether the content should be encoded or decoded")
             .required(true)
-            .allowableValues(ENCODE_MODE, DECODE_MODE)
-            .defaultValue(ENCODE_MODE)
+            .allowableValues(EncodingMode.class)
+            .defaultValue(EncodingMode.ENCODE)
             .build();
 
     public static final PropertyDescriptor ENCODING = new PropertyDescriptor.Builder()
             .name("Encoding")
             .description("Specifies the type of encoding used")
             .required(true)
-            .allowableValues(BASE64_ENCODING, BASE32_ENCODING, HEX_ENCODING)
-            .defaultValue(BASE64_ENCODING)
+            .allowableValues(EncodingType.class)
+            .defaultValue(EncodingType.BASE64_ENCODING)
+            .build();
+
+    static final PropertyDescriptor LINE_OUTPUT_MODE = new PropertyDescriptor.Builder()
+            .name("Line Output Mode")
+            .displayName("Line Output Mode")
+            .description("If set to 'single-line', the encoded FlowFile content will output as a single line. If set to 'multiple-lines', "
+                + "it will output as multiple lines. This property is only applicable when Base64 or Base32 encoding is selected.")
+            .required(false)
+            .defaultValue(LineOutputMode.SINGLE_LINE)
+            .allowableValues(LineOutputMode.class)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(MODE, EncodingMode.ENCODE)
+            .dependsOn(ENCODING, EncodingType.BASE64_ENCODING, EncodingType.BASE32_ENCODING)
             .build();
 
+    static final PropertyDescriptor ENCODED_LINE_SEPARATOR = new PropertyDescriptor.Builder()
+        .name("Encoded Content Line Separator")
+        .displayName("Encoded Content Line Separator")
+        .description("Each line of encoded data will be terminated with this byte sequence (e.g. \\r\\n"
+                + "). This property defaults to the system-dependent line separator string.  If `line-length` <= 0, "
+                + "the `line-separator` property is not used. This property is not used for `hex` encoding.")
+        .required(false)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .defaultValue(System.lineSeparator())
+        .addValidator(Validator.VALID)
+        .dependsOn(MODE, EncodingMode.ENCODE)
+        .dependsOn(ENCODING, EncodingType.BASE64_ENCODING, EncodingType.BASE32_ENCODING)
+        .build();
```

Comment:
I exposed it because it's available in the class.  Need to solicit additional feedback from the community on its usefulness; otherwise can delete.

Reply:
Thanks, I think in this case, the first version of this implementation could hard-code `\n` instead of making it a property. If later on, there is a clear need for this property, it would be added.

Tagging @dan-s1 for additional comment since he has provided feedback thus far.

Reply:
@dan-s1 `Base32OutputStream` and `Base64OutputStream`.  Look for the argument `lineSeparator`.  The thinking at the time was: if it's available in the API, someone might find it useful.

Source: https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/binary/Base32OutputStream.html#%3Cinit%3E(java.io.OutputStream,boolean,int,byte%5B%5D)
Source: https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/binary/Base64OutputStream.html#%3Cinit%3E(java.io.OutputStream,boolean,int,byte%5B%5D,org.apache.commons.codec.CodecPolicy)

Reply:
@knguyen1 Thanks for pointing out those classes.
@exceptionfactory  Based on the above classes there is another property which I am not sure should be exposed or not and that is the decoding policy. There are two options lenient and strict. The default is lenient. The wording from the `Base32OutputStream `Javadoc is

> You can set the decoding behavior when the input bytes contain leftover trailing bits that cannot be created by a valid encoding. These can be bits that are unused from the final character or entire characters. The default mode is lenient decoding.
> 
> Lenient: Any trailing bits are composed into 8-bit bytes where possible. The remainder are discarded.
> Strict: The decoding will raise an [IllegalArgumentException](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html) if trailing bits are not part of a valid encoding. Any unused bits from the final character must be zero. Impossible counts of entire final characters are not allowed.
> When strict decoding is enabled it is expected that the decoded bytes will be re-encoded to a byte array that matches the original, i.e. no changes occur on the final character. This requires that the input bytes use the same padding and alphabet as the encoder.

Reply:
@exceptionfactory [Remove ENCODED_LINE_SEPARATOR property and tests](https://github.com/apache/nifi/pull/8417/commits/ce32323242615a93e70c22a51ba00ce4b7d84a08)